From 41c80295ba2aaec263d372b63a8bb8d9f2846027 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 05 Oct 2018 17:27:07 +0300
Subject: [PATCH] Makefile: release: redefine the RELEASE_BRANCH

RELEASE_BRANCH assumes there is a local branch by the same name as
a remote release branch.
That is not the case for repositories that are not updated often and
forces to create local branches on them too. This is not practical.

Rework the RELEASE_BRANCH definition to get the latest remote release
branch pointing at the current local HEAD.
Here the assumption is that the release branch looks like 'vx.y.z'.

Also, print the release branch name.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index b5f10f8..73d7f5e 100644
--- a/Makefile
+++ b/Makefile
@@ -69,7 +69,11 @@
 
 RELEASE_REPOS := buildroot-external preboot u-boot linux armcp tbot fw_loader btl
 RELEASE_TAG_POINTER := latest-embedded-release
-RELEASE_BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
+RELEASE_BRANCH := $(shell \
+	git branch -r --points-at HEAD | \
+	grep -oE "origin/v[0-9]+\.[0-9]+\.[0-9]+" | \
+	sed 's/origin\///' | sort -n | tail -n 1)
+
 RELEASE_JOB_BRANCH := "package_branch=$(RELEASE_BRANCH)"
 RELEASE_VER_COMPAT := "release_version=$(patsubst v%,%,$(RELEASE_BRANCH))"
 RELEASE_JOB_URL := http://192.168.0.4:8080/job/goya-release-test/buildWithParameters
@@ -93,7 +97,7 @@
 		git tag -f $(RELEASE_TAG_POINTER)
 	@-repo forall $(RELEASE_REPOS) -c git push origin :$(RELEASE_TAG_POINTER)
 	@repo forall $(RELEASE_REPOS) -c git push origin $(HL_FULL_VER) $(RELEASE_TAG_POINTER)
-	@echo "Released: $(HL_FULL_VER)"
+	@echo "Released: $(RELEASE_BRANCH) $(HL_FULL_VER)"
 	@wget -O - "$(RELEASE_JOB_URL)?$(RELEASE_JOB_PARAMS)"
 	@echo "Started the \"goya-release-test\" jenkins job... check your email for results."
 
